gnus-recent-helm.el

```diff
@@ -60,7 +61,9 @@ Helm allows for marked articles or current selection. See
 function `helm-marked-candidates'. Argument _recent is not used."
   (let ((cand (helm-marked-candidates)))
     (dolist (article cand)
-      (cl-delete article gnus-recent--articles-list :test 'equal :count 1))
+      (if (equal article (car gnus-recent--articles-list))
+          (pop gnus-recent--articles-list)
+        (cl-delete article gnus-recent--articles-list :test 'equal :count 1)))
```

Comment:
What's the difference here? Shouldn't cl-delete do the `if equal→pop` for every elt of the list until it gets a hit?

Reply:
That's what I thought !... and got bitten :(
check this simple example:
``(setq tst '(5 8 9 5 10 8  5 23)) 

``(cl-delete 8 tst) --> returns (5 9 5 10 5 23) and
``tst --> is also (5 9 5 10 5 23)

``(cl-delete 5 tst) --> returns (9 10 23), but
``tst  -->  is (5 9 10 23) 

Looks like a bug, doesn't it ?
Fire up SBCL and run the example for 'delete, gives the same result,
Some poor guy, filled [bug#33998]( https://debbugs.gnu.org/cgi/bugreport.cgi?bug=33998) on this issue ;-) 
Quick answer, it is not a bug. The value is changed, not the variable (still points to the same location).
Formally should be doing as (setq gnus-recent--articles-list (cl-delete ...)) or use a construct like the above to avoid an explicit setq on gnus-recent--articles-list.
What's your preference ?

BTW, the other place where the cl-delete is used should be modified similarly too.

Reply:
I originally had `(setq gnus-recent--articles-list (delete` and I think I'd prefer to go with `(setq gnus-recent--articles-list (cl-delete` – it feels less confusing to me at least

Reply:
done